pkg/webhook/policy.go

```diff
@@ -258,72 +258,78 @@ func (h *validationHandler) getValidationMessages(res []*rtypes.Result, req *adm
-			continue
+		actions := r.ScopedEnforcementActions
+		if len(actions) == 0 {
+			actions = []string{r.EnforcementAction}
 		}
-		if *logDenies {
-			h.log.WithValues(
-				logging.Process, "admission",
-				logging.Details, r.Metadata["details"],
-				logging.EventType, "violation",
-				logging.ConstraintName, r.Constraint.GetName(),
-				logging.ConstraintGroup, r.Constraint.GroupVersionKind().Group,
-				logging.ConstraintAPIVersion, r.Constraint.GroupVersionKind().Version,
-				logging.ConstraintKind, r.Constraint.GetKind(),
-				logging.ConstraintAction, r.EnforcementAction,
-				logging.ResourceGroup, req.AdmissionRequest.Kind.Group,
-				logging.ResourceAPIVersion, req.AdmissionRequest.Kind.Version,
-				logging.ResourceKind, req.AdmissionRequest.Kind.Kind,
-				logging.ResourceNamespace, req.AdmissionRequest.Namespace,
-				logging.ResourceName, resourceName,
-				logging.RequestUsername, req.AdmissionRequest.UserInfo.Username,
-			).Info(
-				fmt.Sprintf("denied admission: %s", r.Msg))
-		}
-		if *emitAdmissionEvents {
-			annotations := map[string]string{
-				logging.Process:              "admission",
-				logging.EventType:            "violation",
-				logging.ConstraintName:       r.Constraint.GetName(),
-				logging.ConstraintGroup:      r.Constraint.GroupVersionKind().Group,
-				logging.ConstraintAPIVersion: r.Constraint.GroupVersionKind().Version,
-				logging.ConstraintKind:       r.Constraint.GetKind(),
-				logging.ConstraintAction:     r.EnforcementAction,
-				logging.ResourceGroup:        req.AdmissionRequest.Kind.Group,
-				logging.ResourceAPIVersion:   req.AdmissionRequest.Kind.Version,
-				logging.ResourceKind:         req.AdmissionRequest.Kind.Kind,
-				logging.ResourceNamespace:    req.AdmissionRequest.Namespace,
-				logging.ResourceName:         resourceName,
-				logging.RequestUsername:      req.AdmissionRequest.UserInfo.Username,
+		for _, action := range actions {
+			if err := util.ValidateEnforcementAction(util.EnforcementAction(action), r.Constraint.Object); err != nil {
+				continue
 			}
-			var eventMsg, reason string
-			switch r.EnforcementAction {
-			case string(util.Dryrun):
-				eventMsg = "Dryrun violation"
-				reason = "DryrunViolation"
-			case string(util.Warn):
-				eventMsg = "Admission webhook \"validation.gatekeeper.sh\" raised a warning for this request"
-				reason = "WarningAdmission"
-			default:
-				eventMsg = "Admission webhook \"validation.gatekeeper.sh\" denied request"
-				reason = "FailedAdmission"
+			if *logDenies {
+				h.log.WithValues(
+					logging.Process, "admission",
+					logging.Details, r.Metadata["details"],
+					logging.EventType, "violation",
+					logging.ConstraintName, r.Constraint.GetName(),
+					logging.ConstraintGroup, r.Constraint.GroupVersionKind().Group,
+					logging.ConstraintAPIVersion, r.Constraint.GroupVersionKind().Version,
+					logging.ConstraintKind, r.Constraint.GetKind(),
+					logging.ConstraintAction, r.EnforcementAction,
```

Comment:
are we not logging enforcementactions here?

Reply:
updated to log enforcement actions.